hooks/useColorInterpolation.js

```diff
@@ -0,0 +1,60 @@
+import { useEffect, useRef, useState } from 'react';
+import { Animated } from 'react-native';
+
+const resolveInputRange = colors => {
+  if (colors.length === 1) {
+    // If only one color is given, color will always be the same, no matter of percentage.
+    return [0, 100];
+  }
+
+  const numSteps = colors.length - 1;
+  const stepSize = 100 / numSteps;
+
+  return colors.map((_, index) => Math.round(index * stepSize));
+};
+
+const resolveOutputRange = colors => {
+  if (colors.length === 1) {
+    // If only one color is given, color will always be the same, no matter of percentage.
+    return [colors[0], colors[0]];
+  }
+
+  return colors;
+};
+
+const resolveInterpolationRange = colors => {
+  return {
+    inputRange: resolveInputRange(colors),
+    outputRange: resolveOutputRange(colors),
+  };
+};
+
+export const useColorInterpolation = (colors, percentage) => {
```

Comment:
Based on `GaugeProgressBar` from loyalty, rework the component with Circle base svg component, so that we can animate the progress. Also, create a hook that will return animated progress and current color. We only want to animate one animated value, and do all the calculations, based on one reference. Either create more hooks, or extend this one.

Reply:
percentage -> progressPerecentage